src/main/java/duke/DukeException.java

```diff
@@ -0,0 +1,31 @@
+package duke;
+
+/**
+ * This class encapsulates the Exceptions thrown by the Duke chatbot.
+ *
+ * @author Andrew Lo Zhi Sheng
+ * @version CS2103T AY22/23 Semester 1
+ */
+public class DukeException extends Exception {
+    private String message;
+
+    /**
+     * Constructor for an instance of and exception.
+     *
+     * @param message the error message
+     */
+    public DukeException(String message) {
+        super("OOPS!!!" + message);
+        this.message = message;
```

Comment:
just a suggestion but perhaps you don't have to save the message in a field since you're calling super with the new message already! then you won't have to override toString() method.